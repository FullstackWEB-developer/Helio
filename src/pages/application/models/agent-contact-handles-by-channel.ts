import {ChannelTypes} from '@shared/models';

export interface AgentContactHandlesByChannel {
    agent: string;
    channel: ChannelTypes;
    startInterval?: Date;
    endInterval?: Date;
    contactsHandled?: number
}
