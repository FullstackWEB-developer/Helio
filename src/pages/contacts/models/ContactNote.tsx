export interface ContactNote {
    id?: number;
    noteText: string;
    createdOn?: Date;
    createdBy?: string;
}