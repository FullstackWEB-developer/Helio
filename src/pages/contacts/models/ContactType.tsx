export enum ContactType {
    Individual = 'Individual',
    Company = 'Company'
}