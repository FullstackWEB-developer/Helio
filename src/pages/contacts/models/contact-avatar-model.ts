import {Icon} from '@components/svg-icon/icon';
export interface ContactAvatarModel {
    initials?: string;
    className?: string; 
    isCompany?: boolean;
    iconType?: Icon;
    iconClassName?: string;
    iconFillClass?: string;
}