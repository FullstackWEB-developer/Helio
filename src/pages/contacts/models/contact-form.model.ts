import {Option} from "@components/option/option";

export interface ContactFormModel {
    id?: string;
    firstName?: string;
    lastName?: string;
    companyName?: string;
    category: number | Option;
    jobTitle?: string;
    department?: string;
    email?: string;
    workMainPhone?: string;
    workMainExtension?: string;
    workDirectPhone?: string;
    mobile?: string;
    fax?: string;
    website?: string;
    primaryAddressLine?: string;
    primaryApt?: string;
    primaryCity?: string;
    primaryState?: string | Option;
    primaryZipCode?: string;
    shippingAddressLine?: string;
    shippingApt?: string;
    shippingCity?: string;
    shippingState?: string | Option;
    shippingZipCode?: string;
    billingAddressLine?: string;
    billingApt?: string;
    billingCity?: string;
    billingState?: string | Option;
    billingZipCode?: string;
    isStarred?: boolean;
    relatedId?: string;
}