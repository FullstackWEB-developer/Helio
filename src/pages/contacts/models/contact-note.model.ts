export interface ContactNote {
    id?: string;
    noteText: string;
    createdOn: Date;
    createdByName: string;
    contactId: string;
}
export interface AddContactNoteProps {
    contactId: string;
    contactNoteDto: ContactNote
}