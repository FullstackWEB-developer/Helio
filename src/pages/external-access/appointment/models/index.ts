export * from './appointment-type.model';
export * from './appointment-slot.model';
export * from './appointment.model';
export * from './appointment-cancellation.model';
export * from './appointment-cancel-reason.model';
export * from './appointment-slot-request.model';
