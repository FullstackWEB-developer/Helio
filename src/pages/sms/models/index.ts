export * from "./sms-new-message-steps";
export * from "./sms-notification-data";
export * from "./sms-query-type";
