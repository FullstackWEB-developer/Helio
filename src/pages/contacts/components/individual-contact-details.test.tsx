import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {act, render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import IndividualContactDetails from './individual-contact-details';
import Router from "react-router-dom";
import { Contact, ContactType } from '@shared/models';

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("IndividualContactDetails tests", () => {
    let container: HTMLDivElement | null;
    let mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            },
            auth: {
                name: "Test"
            }
        },
        lookupsState: {

        },
        ticketState: {
            lookupValues: [{
                label: "Test",
                parentValue: "Test"
            },{
                label: "Test",
                parentValue: "Test"
            }]
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        ccpState: {
            chatCounter: 1,
            voiceCounter: 2
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders IndividualContactDetails isEdit true correctly", async () => {
        let contact = {
            category: 1,
            companyName: "",
            firstName: "",
            id: "",
            lastName: "",
            mobilePhone: "",
            name: "",
            type: ContactType.Individual,
            workMainPhone: "",
        } as Contact;
        jest.spyOn(Router, 'useParams').mockReturnValue({ id: '1234' });
        let fragment: any;
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <IndividualContactDetails onUpdateSuccess={() => {
                }} contact={contact} editMode={true}/>
            </TestWrapper>);
            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });

    it("renders IndividualContactDetails isEdit false correctly", async () => {
        let contact = {
            category: 1,
            companyName: "",
            firstName: "",
            id: "",
            lastName: "",
            mobilePhone: "",
            name: "",
            type: ContactType.Individual,
            workMainPhone: "",
        } as Contact;
        jest.spyOn(Router, 'useParams').mockReturnValue({ id: '1234' });
        let fragment: any;
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <IndividualContactDetails onUpdateSuccess={() => {
                }} contact={contact} editMode={false}/>
            </TestWrapper>);
            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });
})
