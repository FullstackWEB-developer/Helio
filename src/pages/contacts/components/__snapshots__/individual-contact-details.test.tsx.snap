// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndividualContactDetails tests renders IndividualContactDetails isEdit false correctly 1`] = `
<DocumentFragment>
  <div
    class="grid grid-cols-8 gap-2 body2"
  >
    <div
      class="contact-light "
    >
      contacts.contact_details.individual.company
    </div>
    <div
      class="pl-10 flex whitespace-pre-line col-span-7"
    >
      <div
        class=""
        data-testid="contact-info-field-contacts.contact_details.individual.company"
      >
        common.not_available
      </div>
    </div>
    <div
      class="contact-light "
    >
      contacts.contact_details.individual.category
    </div>
    <div
      class="pl-10 flex whitespace-pre-line col-span-7"
    >
      <div
        class=""
        data-testid="contact-info-field-contacts.contact_details.individual.category"
      />
    </div>
    <div
      class="contact-light "
    >
      contacts.contact_details.individual.department
    </div>
    <div
      class="pl-10 flex whitespace-pre-line col-span-7"
    >
      <div
        class=""
        data-testid="contact-info-field-contacts.contact_details.individual.department"
      >
        common.not_available
      </div>
    </div>
    <div
      class="contact-light "
    >
      contacts.contact_details.individual.email
    </div>
    <div
      class="pl-10 flex whitespace-pre-line col-span-7"
    >
      <div
        class="pr-1"
      >
        <div
          class=""
        >
          <svg
            class=" fill-default"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              opacity="0.7"
              width="24"
            />
            <path
              class="rgba-025-fill"
              d="M18.125,64H1.875A1.875,1.875,0,0,0,0,65.875v11.25A1.875,1.875,0,0,0,1.875,79h16.25A1.875,1.875,0,0,0,20,77.125V65.875A1.875,1.875,0,0,0,18.125,64Zm0,1.875v1.594c-.876.713-2.272,1.822-5.257,4.16-.658.517-1.961,1.761-2.868,1.746-.907.015-2.21-1.229-2.868-1.746-2.985-2.337-4.381-3.446-5.257-4.16V65.875ZM1.875,77.125v-7.25c.9.713,2.164,1.713,4.1,3.228.854.672,2.349,2.156,4.026,2.147,1.669.009,3.145-1.453,4.026-2.146,1.935-1.515,3.2-2.516,4.1-3.229v7.25Z"
              transform="translate(2 -59)"
            />
          </svg>
        </div>
      </div>
      <div
        class=""
        data-testid="contact-info-field-contacts.contact_details.individual.email"
      >
        common.not_available
      </div>
    </div>
    <div
      class="contact-light "
    >
      contacts.contact_details.individual.work_main_phone
    </div>
    <div
      class="pl-10 flex whitespace-pre-line col-span-7"
    >
      <div
        class="pr-1"
      >
        <div
          class=""
        >
          <svg
            class=" fill-default"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="rgba-025-fill"
              d="M17.336.854,13.68.011A.849.849,0,0,0,12.713.5L11.025,4.437a.842.842,0,0,0,.243.984L13.4,7.165a13.029,13.029,0,0,1-6.23,6.23l-1.744-2.13a.843.843,0,0,0-.984-.243L.5,12.709a.854.854,0,0,0-.492.97l.844,3.656a.843.843,0,0,0,.823.654A16.31,16.31,0,0,0,17.99,1.677.843.843,0,0,0,17.336.854Z"
              transform="translate(3.011 3.011)"
            />
          </svg>
        </div>
      </div>
      <div
        class=""
        data-testid="contact-info-field-contacts.contact_details.individual.work_main_phone"
      >
        common.not_available
      </div>
    </div>
    <div
      class="contact-light "
    >
      contacts.contact_details.individual.work_direct_phone
    </div>
    <div
      class="pl-10 flex whitespace-pre-line col-span-7"
    >
      <div
        class="pr-1"
      >
        <div
          class=""
        >
          <svg
            class=" fill-default"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="rgba-025-fill"
              d="M17.336.854,13.68.011A.849.849,0,0,0,12.713.5L11.025,4.437a.842.842,0,0,0,.243.984L13.4,7.165a13.029,13.029,0,0,1-6.23,6.23l-1.744-2.13a.843.843,0,0,0-.984-.243L.5,12.709a.854.854,0,0,0-.492.97l.844,3.656a.843.843,0,0,0,.823.654A16.31,16.31,0,0,0,17.99,1.677.843.843,0,0,0,17.336.854Z"
              transform="translate(3.011 3.011)"
            />
          </svg>
        </div>
      </div>
      <div
        class=""
        data-testid="contact-info-field-contacts.contact_details.individual.work_direct_phone"
      >
        common.not_available
      </div>
    </div>
    <div
      class="contact-light "
    >
      contacts.contact_details.individual.mobile_phone
    </div>
    <div
      class="pl-10 flex whitespace-pre-line col-span-7"
    >
      <div
        class="pr-1"
      >
        <div
          class=""
        >
          <svg
            class=" fill-default"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="rgba-025-fill"
              d="M17.336.854,13.68.011A.849.849,0,0,0,12.713.5L11.025,4.437a.842.842,0,0,0,.243.984L13.4,7.165a13.029,13.029,0,0,1-6.23,6.23l-1.744-2.13a.843.843,0,0,0-.984-.243L.5,12.709a.854.854,0,0,0-.492.97l.844,3.656a.843.843,0,0,0,.823.654A16.31,16.31,0,0,0,17.99,1.677.843.843,0,0,0,17.336.854Z"
              transform="translate(3.011 3.011)"
            />
          </svg>
        </div>
      </div>
      <div
        class=""
        data-testid="contact-info-field-contacts.contact_details.individual.mobile_phone"
      >
        common.not_available
      </div>
    </div>
    <div
      class="contact-light "
    >
      contacts.contact_details.individual.fax
    </div>
    <div
      class="pl-10 flex whitespace-pre-line col-span-7"
    >
      <div
        class="pr-1"
      >
        <div
          class=""
        >
          <svg
            class=" "
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="rgba-025-fill"
              d="M17.336.854,13.68.011A.849.849,0,0,0,12.713.5L11.025,4.437a.842.842,0,0,0,.243.984L13.4,7.165a13.029,13.029,0,0,1-6.23,6.23l-1.744-2.13a.843.843,0,0,0-.984-.243L.5,12.709a.854.854,0,0,0-.492.97l.844,3.656a.843.843,0,0,0,.823.654A16.31,16.31,0,0,0,17.99,1.677.843.843,0,0,0,17.336.854Z"
              transform="translate(3.011 3.011)"
            />
          </svg>
        </div>
      </div>
      <div
        class=""
        data-testid="contact-info-field-contacts.contact_details.individual.fax"
      >
        common.not_available
      </div>
    </div>
    <div
      class="contact-light "
    >
      contacts.contact_details.individual.website
    </div>
    <div
      class="pl-10 flex whitespace-pre-line col-span-7"
    >
      <div
        class=""
        data-testid="contact-info-field-contacts.contact_details.individual.website"
      >
        common.not_available
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`IndividualContactDetails tests renders IndividualContactDetails isEdit true correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col relative overflow-hidden"
  >
    <form
      novalidate=""
    >
      <div
        class="grid grid-cols-1 lg:grid-cols-12 gap-x-8"
      >
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-first-name"
                data-testid="firstName"
                name="firstName"
                placeholder=""
                required=""
                type="text"
                value=""
              />
              <label
                class="absolute truncate required body2 "
              >
                contacts.contact_details.individual.first_name
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-last-name"
                data-testid="lastName"
                name="lastName"
                placeholder=""
                required=""
                type="text"
                value=""
              />
              <label
                class="absolute truncate required body2 "
              >
                contacts.contact_details.individual.last_name
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-company-name"
                data-testid="companyName"
                name="companyName"
                placeholder=""
                type="text"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.company
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="select-wrapper relative w-full flex flex-col h-20 "
          >
            <div
              class="select relative flex flex-col "
            >
              <input
                autocomplete="nope"
                autocorrect="off"
                class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
                data-testid="category"
                name="category"
                readonly=""
                spellcheck="false"
                type="text"
                value=""
              />
              <label
                class="select-label absolute truncate"
              >
                <span
                  class="text-danger"
                >
                  *
                </span>
                <span
                  class="select-label-span body2"
                >
                  contacts.contact_details.individual.category
                </span>
              </label>
              <div
                class="absolute pt-4 right-4 "
              >
                <div
                  class="flex flex-row items-start"
                >
                  <div
                    class="cursor-pointer"
                  >
                    <div
                      class=""
                    >
                      <svg
                        class="icon-medium "
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="none"
                          height="24"
                          width="24"
                        />
                        <path
                          class="select-arrow-fill"
                          d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                          transform="translate(5 16.273) rotate(-90)"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="absolute py-2 options"
              >
                <div
                  class="w-full pt-2 text-center subtitle3-small"
                >
                  common.autocomplete_search
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-job-title"
                data-testid="jobTitle"
                name="jobTitle"
                placeholder=""
                type="text"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.job_title
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-department"
                data-testid="department"
                name="department"
                placeholder=""
                type="text"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.department
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-email"
                data-testid="email"
                name="email"
                placeholder=""
                type="email"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.email
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-3"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-work-main-phone"
                data-testid="workMainPhone"
                name="workMainPhone"
                placeholder=""
                type="tel"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.work_main_phone
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-2"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-work-main-extension"
                data-testid="workMainExtension"
                name="workMainExtension"
                placeholder=""
                type="number"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.extension
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-work_direct_phone"
                data-testid="workDirectPhone"
                name="workDirectPhone"
                placeholder=""
                type="tel"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.work_direct_phone
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-mobile"
                data-testid="mobile"
                name="mobile"
                placeholder=""
                type="tel"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.mobile
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-fax"
                data-testid="fax"
                name="fax"
                placeholder=""
                type="tel"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.fax
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-website"
                data-testid="website"
                name="website"
                placeholder=""
                type="website"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.website
              </label>
            </div>
          </div>
        </div>
      </div>
      <div
        class="h-10 flex items-center mt-4 mb-2"
      >
        contacts.contact_details.individual.address
      </div>
      <div
        class="grid grid-cols-1 lg:grid-cols-12 gap-x-8"
      >
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-primary-address-line"
                data-testid="primaryAddressLine"
                name="primaryAddressLine"
                placeholder=""
                type="text"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.address
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contactprimary-address-apt"
                data-testid="primaryApt"
                name="primaryApt"
                placeholder=""
                type="text"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.apt
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-5"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-primary-address-city"
                data-testid="primaryCity"
                name="primaryCity"
                placeholder=""
                type="text"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.city
              </label>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-3"
        >
          <div
            class="select-wrapper relative w-full flex flex-col h-20 "
          >
            <div
              class="select relative flex flex-col "
            >
              <input
                autocomplete="nope"
                autocorrect="off"
                class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
                data-testid="primaryState"
                name="primaryState"
                readonly=""
                spellcheck="false"
                type="text"
                value=""
              />
              <label
                class="select-label absolute truncate"
              >
                <span
                  class="select-label-span body2"
                >
                  contacts.contact_details.individual.state
                </span>
              </label>
              <div
                class="absolute pt-4 right-4 "
              >
                <div
                  class="flex flex-row items-start"
                >
                  <div
                    class="cursor-pointer"
                  >
                    <div
                      class=""
                    >
                      <svg
                        class="icon-medium "
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="none"
                          height="24"
                          width="24"
                        />
                        <path
                          class="select-arrow-fill"
                          d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                          transform="translate(5 16.273) rotate(-90)"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="absolute py-2 options"
              >
                <div
                  class="w-full pt-2 text-center subtitle3-small"
                >
                  common.autocomplete_search
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 lg:col-span-2"
        >
          <div
            class="flex flex-col h-20 input-group"
          >
            <div
              class="input-group-container flex flex-wrap items-stretch w-full relative  "
            >
              <input
                autocomplete="off"
                class=" pt-6 body2 h-14 flex-shrink flex-grow flex-auto leading-normal w-px flex-1 input-date-mask pl-4"
                data-test-id="contact-primary-zip-code"
                data-testid="primaryZipCode"
                maxlength="5"
                name="primaryZipCode"
                placeholder=""
                type="text"
                value=""
              />
              <label
                class="absolute truncate  body2 "
              >
                contacts.contact_details.individual.zip_code
              </label>
            </div>
          </div>
        </div>
      </div>
      <div
        class="relative py-2.5 cursor-pointer"
      >
        <div
          class="flex items-center"
        >
          <div
            class=""
          >
            <svg
              class="icon-medium "
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
              />
              <path
                class="fill-default"
                d="M19,13H13v6H11V13H5V11h6V5h2v6h6Z"
              />
            </svg>
          </div>
          <span
            class="body2 pl-2 contact-accent-color"
          >
            contacts.new-contact.add_more
          </span>
        </div>
      </div>
      <div
        class="grid grid-cols-1 lg:grid-cols-12 gap-x-8"
      >
        <div
          class="col-span-12 lg:col-span-10"
        >
          <div
            class="flex items-center justify-center h-20 mb-4 full-w"
          >
            <button
              class="secondary-button h-10 secondary-contact-form-btn mr-8"
              type="button"
            >
              <div
                class="flex flex-row justify-center items-center space-x-2"
              >
                <div>
                  common.cancel
                </div>
              </div>
            </button>
            <button
              class="medium-button"
              disabled=""
              type="submit"
            >
              <div
                class="flex flex-row justify-center items-center space-x-2"
              >
                <div>
                  common.save
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </form>
    <div
      class="flex items-center justify-center m-auto justify-self-center"
    />
  </div>
</DocumentFragment>
`;
